opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -35,42 +39,65 @@ impl<T: Number<T>> ValueMap<T> {
             values: RwLock::new(HashMap::new()),
             has_no_value_attribute_value: AtomicBool::new(false),
             no_attribute_value: T::new_atomic_tracker(),
+            count: AtomicUsize::new(0),
         }
     }
 }
 
 impl<T: Number<T>> ValueMap<T> {
-    fn measure(&self, measurement: T, attrs: AttributeSet) {
+    fn measure(&self, measurement: T, attrs: &[KeyValue]) {
```

Comment:
this is the key - we operate on incoming ref/slice itself **without allocating, copying, sorting, de-duplicating**. If the look up fails, the "slow" path will do everything as before. The net effect is, for users who use metrics API normally (i.e without duplicates, and without changing the order of attributes), we'll have highest performance and zero heap allocations.